### ISSUE TITLE
@State/@Binding does not work properly with structs

### ISSUE THREAD

**Issue by pnewell** [skiptools/skip-ui#8]
I am reporting a bug with @State/@Binding implementation. When binding to a field of a struct stored with @State, the @Binding does not properly update the struct.

Here is a basic example. This works great on iOS but does not work on Android.

```
struct Item {
  let id: UUID
  var title: String
}

struct ObservablePlayground: View {
    @State var item = Item(id: UUID(), title: "New Item")
    var body: some View {
        VStack {
            Text(item.title)
            BindingView(text: $item.title)
                .border(Color.blue)
        }
    }
}

struct BindingView: View {
    @Binding var text: String
    var body: some View {
        Button("Button") {
            text = "bindingState"
        }
        .accessibilityIdentifier("binding-button")
    }
}
```

**Comment by aabewhite** [comment]
Thank you for the detailed report! Reproduced. We'll work on a fix.

**Comment by aabewhite** [comment]
Fixed in skip 0.7.28 and skip-ui 0.3.16. Use Xcode->File->Packages->Update to Latest Package Versions and you'll need to do a clean rebuild. Please let us know if you have any problems.
